BUG: axis kw not propogating on pct_change

Tux1 opened the pull request:
axis option was not correctly set on fillna in pct_change

Tux1 committed:
Update generic.py

axis option was not correctly set on fillna

Tux1 committed:
BUG: axis kw not propogating on pct_change #11150

test regarding buf fix on pct_change propagation axis

revamping test_pct_change. Rename, multi axis and error fixed

sinhrks commented:
Thanks. Can you add a test for this?

jreback commented:
@Tux1 pls add a test for this

Tux1 commented:
Sorry, it is my first time.
Is it okay ? I pushed a simple test that shows the issue and my previous commit fixed it.
Thanks

Tux1 commented:
Sorry. It is my first commit.
I made a simple test that shows the issue and my commit fixed it.
Is it okay ?
Thanks

jreback commented:
pls see the contributing docs: http://pandas.pydata.org/pandas-docs/stable/contributing.html

tests / fix should be on a single pr.

Tux1 committed:
test regarding buf fix on pct_change propagation axis

Tux1 commented:
Sorry... Moreover the test was wrong. 
I have made another test which is passed with my patch. That is committed in a single branch, this one.

jreback commented on pandas/tests/test_generic.py:
```diff
@@ -1851,6 +1851,15 @@ def test_pipe_panel(self):
             result = wp.pipe((f, 'y'), x=1, y=1)
 
+    def test_pct_change(self):
+        pnl = DataFrame([np.arange(0, 40, 10), np.arange(0, 40, 10), np.arange(0, 40, 10)]).astype(np.float64)
```
add the issue number here

jreback commented:
pls add a whatsnew note for 0.17.1 (bug fix)

jreback commented on pandas/tests/test_generic.py:
```diff
@@ -1851,6 +1851,15 @@ def test_pipe_panel(self):
+        pnl.iat[1,0] = np.nan
+        pnl.iat[1,1] = np.nan
+        
+        expected = pnl.ffill(axis=1).pct_change(axis=1, fill_method=None)
```
test with axis 0, 2 as well (if their are existing tests for that, pls move them here)

further this should be a bit more generic (or append _frame) to the test name as the ``generic.py`` tests should test Series,DataFrame,Panel

rekcahpassyla committed:
BUG: Fixes GH11372

Add an optional argument to _NDFrameIndexer._align_series to indicate
if the indexer is from a MultiIndex

jreback commented:
@Tux1 can you rebase / update

Tux1 committed:
revamping test_pct_change. Rename, multi axis and error fixed

Tux1 committed:
Merge branch 'patch-1' of https://github.com/Tux1/pandas into patch-1

Tux1 committed:
Merge remote-tracking branch 'pydata/master' into patch-1

Tux1 committed:
BUG: axis kw not propogating on pct_change #11150

test_pct_change with loop over axis

backtick added

Tux1 committed:
BUG: axis kw not propogating on pct_change #11150

Tux1 commented:
Sorry for several commits. I think that is okay now ? whatsnew edited, commit rebased

cfperez committed:
BUG: implement Series.sort_index(..inplace=True) for #11402

jreback commented on doc/source/whatsnew/v0.17.1.txt:
```diff
@@ -96,7 +96,7 @@ Bug Fixes
 
 
-
+- Bug in ``DataFrame.pct_change()`` was not propagating axis on fillna method (:issue:`11150`)
```
use double backticks around ``axis`` (and say ``axis`` keyword) and 'on ``.fillna()`` method'

jreback commented on pandas/tests/test_generic.py:
```diff
@@ -1859,6 +1859,26 @@ def test_pipe_panel(self):
             result = wp.pipe((f, 'y'), x=1, y=1)
 
+    def test_pct_change_frame(self):
+        pnl = DataFrame([np.arange(0, 40, 10), np.arange(0, 40, 10), np.arange(0, 40, 10)]).astype(np.float64)
```
add the issue number as a comment here

jreback commented on pandas/tests/test_generic.py:
```diff
@@ -1859,6 +1859,26 @@ def test_pipe_panel(self):
+        result_axis0 = pnl.pct_change(axis=0, fill_method='pad')
+
+        self.assert_frame_equal(result_axis0, expected_axis0)
+
```
move this to ``test_frame`` right after the other ``test_pct_change_*`` tests

Tux1 replied:
what do you mean ? There is no ``test_frame`` and I made the only one ``test_pct_change_frame`` test

jreback commented on pandas/tests/test_generic.py:
```diff
@@ -1859,6 +1859,26 @@ def test_pipe_panel(self):
+        
+        mask = pnl.isnull()
+
+        expected_axis0 = pnl.ffill(axis=0)/pnl.ffill(axis=0).shift(axis=0) - 1
```
you can do this in a loop over both axes

jreback commented:
some comments, pls squash when finished

jreback committed:
CI: use latest for 3.5 build (e.g. numpy 1.10.1), #11187

jreback committed:
TST: invalid test for unsortable for > numpy 1.10, #11138

Tux1 committed:
duplicate in whatsnew deleted and typing erro fixed

jreback commented:
pls rebase on master / squash, see docs [here](http://pandas.pydata.org/pandas-docs/stable/contributing.html#contributing-your-changes-to-pandas)

jreback commented on pandas/tests/test_generic.py:
```diff
@@ -1859,6 +1859,23 @@ def test_pipe_panel(self):
+    def test_pct_change_frame(self):
+        pnl = DataFrame([np.arange(0, 40, 10), np.arange(0, 40, 10), np.arange(0, 40, 10)]).astype(np.float64)
+        pnl.iat[1,0] = np.nan
+        pnl.iat[1,1] = np.nan
```
I mean that this function needs to move to ``tests\test_frame.py`` (and not in ``test_generic.py``)

Tux1 committed:
BUG: axis kw not propogating on pct_change #11150

Tux1 commented:
Sorry, my repo is a mess... I will be better next time

jreback commented:
merged via https://github.com/pydata/pandas/commit/334b07641a1dc8c1e7390d01184e114ddde0a7da

thanks!
